File: client/GhostClient/Core/Network/Packets/SpawnEffectPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class SpawnEffectPacket : Packet
    {
        public SpawnEffectPacket(Stream stream) : base(stream)
        {
        }

        public SpawnEffectPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public SpawnEffectPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            byte effectType = Consume(1).ByteValue;
            int duration = Consume(4).IntValue;
            int size = Consume(4).IntValue;
            float x = Consume(4).FloatValue;
            float y = Consume(4).FloatValue;

            double rotation = Consume(8).DoubleValue;

            GameHandler.Effects[effectType].Begin(duration, size, x, y, rotation);
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/SpawnEffectPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.core.sprites.effects.Effect
import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class SpawnEffectPacket : Packet<PlayerClient>() {

    override fun handle() {
        val effectType = consume(1).asByte()
        val duration = consume(4).asInt()
        val size = consume(4).asInt()
        val x = consume(4).asFloat()
        val y = consume(4).asFloat()

        val rotation = consume(8).asDouble()

        Effect.EFFECTS[effectType.toInt()].begin(duration, size, x, y, rotation)
    }
}
